### PR TITLE
Refactor monitoring code

### PR DESCRIPTION
**Pull request by avlitman**
**What this PR does / why we need it**:
Following the design doc: https://github.com/kubevirt/community/pull/219
and the initial work in KubeVirt core: https://github.com/kubevirt/kubevirt/pull/10044

this PR refactors the monitoring package to uniformize it across all KubeVirt components

**Which issue(s) this PR fixes**:
https://issues.redhat.com/browse/CNV-28497

**Special notes for your reviewer**:

**Release note**:

```
Refactor monitoring code

```

### CODE REVIEW

**Comment by avlitman** [comment]
@awels This pr is verified manually (checked that the metrics work and have correct initial values).
 Thanks!

File: pkg/monitoring/metrics/operator_metrics.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+var (
+	operatorMetrics = []operatormetrics.Metric{
+		readyGauge,
+	}
+
+	readyGauge = operatormetrics.NewGauge(
+		operatormetrics.MetricOpts{
+			Name: "kubevirt_hpp_cr_ready",
+			Help: "The number of running hostpath-provisioner-operator pods",
```
Comment (awels) [inline pkg/monitoring/metrics/operator_metrics.go:31]: That help doesn't make sense, this is about if the HPP CR status reports ready or not. Not the number of hpp pods.

**Review by awels — COMMENTED** [review]
looks good, just one minor comment

File: pkg/monitoring/metrics/operator_metrics.go (hunk 1 above)
Comment (avlitman) [reply pkg/monitoring/metrics/operator_metrics.go]: You are right I replaced the help by mistake. 
@awels fixed now

**Review by awels — APPROVED** [review]

**Pull request merged**